rmgpy/tools/canteramodel.py

```diff
@@ -208,24 +241,36 @@ def __init__(self, species_list=None, reaction_list=None, canteraFile='', output
         `sensitive_species`: a list of RMG species objects for conductng sensitivity analysis on
         `thermo_SA`: a boolean indicating whether or not to run thermo SA. By default, if sensitive_species is given,
                       only kinetic_SA will be calculated and it must be additionally specified to perform thermo SA.
+        `surface_species_list`: list of RMG species objects contianing the surface species. This is necessary for all surface mechanisms to initialize the starting mole fractions.
+        `surface_reaction_list`: list of RMG reaction objects containing the surface reactions.
+        For surface mechanisms, either canteraFile must be provided, or load_chemkin_model() must be called later on to create the Cantera file with the surface mechanism.
         """
         self.species_list = species_list
         self.reaction_list = reaction_list
         self.reaction_map = {}
         self.model = ct.Solution(canteraFile) if canteraFile else None
+        self.surface = bool(surface_species_list or surface_reaction_list)
+        self.surface_species_list = surface_species_list
+        self.surface_reaction_list = surface_reaction_list
         self.output_directory = output_directory if output_directory else os.getcwd()
         self.conditions = conditions if conditions else []
         self.sensitive_species = sensitive_species if sensitive_species else []
         self.thermo_SA = thermo_SA
 
+        if self.surface:
+            assert surface_species_list, "Surface species list must be specified to run a surface simulation"
+            if canteraFile:
+                self.surface = ct.Interface(canteraFile, 'surface1')
+                self.model = self.surface.adjacent['gas']
+
         # Make output directory if it does not yet exist:
         if not os.path.exists(self.output_directory):
             try:
                 os.makedirs(self.output_directory)
             except:
                 raise Exception('Cantera output directory could not be created.')
 
-    def generate_conditions(self, reactor_type_list, reaction_time_list, mol_frac_list, Tlist=None, Plist=None, Vlist=None):
+    def generate_conditions(self, reactor_type_list, reaction_time_list, mol_frac_list, surface_mol_frac_list=[], Tlist=None, Plist=None, Vlist=None):
```

Comment:
Same here, can you use None as default and initialize the list within the function